arm-redis/2017-02-01/swagger/redis.json

```diff
@@ -0,0 +1,1462 @@
+        },
+        "location": {
+          "type": "string",
+          "description": "Resource location."
+        },
+        "tags": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Resource tags."
+        }
+      },
+      "required": [
+        "location"
+      ],
+      "x-ms-azure-resource": true
+    },
+    "RedisCreateParameters": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/RedisCreateProperties",
+          "description": "Redis cache properties."
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "Parameters supplied to the Create Redis operation."
+    },
+    "RedisUpdateParameters": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/RedisUpdateProperties",
+          "description": "Redis cache properties."
+        }
+      },
+      "description": "Parameters supplied to the Update Redis operation."
+    },
+    "RedisAccessKeys": {
+      "properties": {
+        "primaryKey": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The current primary key that clients can use to authenticate with Redis cache."
+        },
+        "secondaryKey": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The current secondary key that clients can use to authenticate with Redis cache."
+        }
+      },
+      "description": "Redis cache access keys."
+    },
+    "RedisResourceProperties": {
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "The SKU of the Redis cache to deploy."
+        },
+        "redisVersion": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Redis version."
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Redis instance provisioning status."
+        },
+        "hostName": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Redis host name."
+        },
+        "port": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "description": "Redis non-SSL port."
+        },
+        "sslPort": {
+          "readOnly": true,
+          "type": "integer",
+          "format": "int32",
+          "description": "Redis SSL port."
+        },
+        "accessKeys": {
+          "$ref": "#/definitions/RedisAccessKeys",
+          "readOnly": true,
+          "description": "The keys of the Redis cache - not set if this object is not the response to Create or Update redis cache"
+        },
+        "linkedServers": {
```

Comment:
Instead of returning the complete object, you should just return the Ids of the linked server. Such a model can lead to Escalation Of Privilege where someone who does not have read access to the linked server but only has access to the redis cache, will be able to read the linked server resource and its properties.